lib/danger/request_sources/bitbucket_server_api.rb

```diff
@@ -63,9 +63,17 @@ def post(uri, body)
         req = Net::HTTP::Post.new(uri.request_uri, { "Content-Type" => "application/json" })
         req.basic_auth @username, @password
         req.body = body
-        Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
           http.request(req)
         end
+
+        # show error to the user when BitBucket Server returned an error
+        case res
+        when Net::HTTPClientError, Net::HTTPServerError
+          # HTTP 4xx - 5xx
+          puts "\nDANGER > Error posting comment to BitBucket Server: #{res.code} (#{res.message})\n\n"
```

Comment:
[Cork](https://github.com/cocoapods/cork) is our UI library :)

I'm not sure we've DI'd in a cork to the APIs, I think the UI layer tends to sit behind it. When a GitHub comment fails, we can still continue because we still can use the status API, I'm not sure if this is the same for BitBucket - if the comment is Danger's only way of communicating, I think @JuanitoFatas' `abort` probably makes the most sense. As it will also make Danger give an error status code.